### PR TITLE
Dataset view - wires up search bar and formats metadata

### LINKED ISSUE #38: View a dataset: Conduct a search

### PR DESCRIPTION
**Pull request by hannako**
closes #38 and #40 

DONE
- expected update works
- location in metadata is formatted 
- search bar works
- search tips page

TO DO
- more specs
- time series data grouping

### CODE REVIEW

File: app/helpers/datasets_helper.rb [hunk 1]
```diff
@@ -23,4 +36,26 @@ def documentation?(key)
     key != nil && key != ""
   end
 
+  def expected_update
+    last = Time.parse(@dataset["updated_at"])
```
Comment (rossjones) [inline app/helpers/datasets_helper.rb]: I think if we use updated_at a change in say, summary, will result in this being wrong.  I think it should really use the end_date of the most recent 'link' to be accurate.

File: app/helpers/datasets_helper.rb [hunk 2]
```diff
@@ -23,4 +36,47 @@ def documentation?(key)
     key != nil && key != ""
   end
 
+  def expected_update
+    if @dataset["datafiles"].any?
+      datafile_next_updated(@dataset["frequency"])
+    else
+      # TODO placeholder error message - should we just hide field if there are no datafiles?
+      "<dd class='unavailable'> This dataset has no data yet </dd>".html_safe
```
Comment (govuklaurence) [inline app/helpers/datasets_helper.rb]: Move the html out to the view and have it depend on an instance var

File: app/helpers/datasets_helper.rb [hunk 3]
```diff
@@ -23,4 +36,47 @@ def documentation?(key)
...
+    end
+  end
+
+  def datafile_next_updated(freq)
+    last = Time.parse(most_recent_date)
+
+    if FREQUENCIES.has_key?(freq)
+      "<dd>#{ last.advance(FREQUENCIES[freq]).strftime("%d %B %Y") }</dd>".html_safe
```
Comment (govuklaurence) [inline app/helpers/datasets_helper.rb]: Same deal here

File: app/helpers/datasets_helper.rb (hunk 2 above)
Comment (hannako) [inline app/helpers/datasets_helper.rb]: https://waffle.io/datagovuk/find_data_beta/cards/597f49f06e469a00619c7916

File: app/helpers/datasets_helper.rb (hunk 3 above)
Comment (hannako) [inline app/helpers/datasets_helper.rb]: https://waffle.io/datagovuk/find_data_beta/cards/597f49f06e469a00619c7916

**Pull request merged**